lib/Conversion/FIRRTLToHW/LowerToHW.cpp

```diff
@@ -4490,37 +4526,42 @@ LogicalResult FIRRTLLowering::lowerVerificationStatement(
         assumeLabel = StringAttr::get(builder.getContext(),
                                       "assume__" + label.getValue());
       addToIfDefBlock("USE_PROPERTY_AS_CONSTRAINT", [&]() {
-        builder.create<sv::AssumeConcurrentOp>(
-            circt::sv::EventControlAttr::get(builder.getContext(), event),
-            clock, predicate, assumeLabel);
+        if (!isUnrOnlyAssert) {
+          builder.create<sv::AssumeConcurrentOp>(
+              circt::sv::EventControlAttr::get(builder.getContext(), event),
+              clock, predicate, assumeLabel);
+        } else {
+          builder.create<sv::AlwaysOp>(
+              ArrayRef(sv::EventControl::AtEdge), ArrayRef(predicate), [&]() {
+                buildImmediateVerifOp(builder, "assume", predicate,
+                                      circt::sv::DeferAssertAttr::get(
+                                          builder.getContext(),
+                                          circt::sv::DeferAssert::Immediate),
+                                      assumeLabel);
+              });
+        }
       });
     }
   };
 
   // Wrap the verification statement up in the optional preprocessor
   // guards. This is a bit awkward since we want to translate an array of
-  // guards into a recursive call to `addToIfDefBlock`.
-  ArrayRef<Attribute> guards{};
-  if (auto guardsAttr = op->template getAttrOfType<ArrayAttr>("guards"))
-    guards = guardsAttr.getValue();
-  bool anyFailed = false;
+  // guards (StringRefs) into a recursive call to `addToIfDefBlock`.
+
+  // Explicitly use a new variable to hold the ref and use that to recurse /
+  // mutate. Does not create any new array - points to the same underlying data.
+  // Hence should not impact performance.
+  auto guardsStringRefCopy = guardsStringRef;
```

Comment:
@seldridge  please note the reason why I touched this logic is basically to avoid duplication of code - and reuse the `guardsStringRef` which is what is needed by this section.

Reply:
As a result of this change - this section became slightly cleaner and hence a bunch of code got removed.

Reply:
I'll usually structure these type of NFC changes _after_ the PR to avoid noise.

Reply:
@seldridge soudns good!

Reply:
Will remove this specific change from the PR.